src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,15 +1,70 @@
 package wolox.training.controllers;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
 @Controller
 public class BookController {
+    @Autowired
+    private BookRepository bookRepository;
+
     @GetMapping("/greeting")
     public String greeting(@RequestParam(name="name", required=false, defaultValue="World") String name, Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
+
+    @GetMapping
+    public Iterable findAll() {
+        return bookRepository.findAllBooks();
+    }
+
+    @GetMapping("/title/{bookAuthor}")
+    public Book findOneByAuthor(@PathVariable String bookAuthor) {
+        return bookRepository.findFirstByAuthor(bookAuthor)
```

Comment:
Use final here and in all params

Reply:
in all the methods?

Reply:
yes in variables and params where the value won't change. You can read the advantages of use final from this answer from stackoverflow https://softwareengineering.stackexchange.com/questions/98691/excessive-use-final-keyword-in-java that is very accurate.